contracts/interchain-token-service/src/abi.rs

```diff
@@ -258,12 +259,67 @@ mod tests {
     }
 
     #[test]
-    #[should_panic(expected = "Invalid UTF-8 sequence")]
-    fn to_std_string_panics_invalid_utf8_bytes() {
+    fn soroban_str_to_std_string() {
         let env = Env::default();
-        let invalid_sequence = vec![0x80, 0x81, 0x82, 0x83];
-        let invalid_string = String::from_bytes(&env, &invalid_sequence);
-        to_std_string(invalid_string);
+
+        let plain_string = "hello world";
+        let plain_string_soroban = String::from_str(&env, &plain_string);
+        assert_eq!(
+            to_std_string(plain_string_soroban).unwrap(),
+            StdString::from(plain_string)
+        );
+
+        let var_length_chars = "🎉中🚀π🌈€";
+        let var_length_chars_soroban = String::from_str(&env, &var_length_chars);
+        assert_eq!(
+            to_std_string(var_length_chars_soroban).unwrap(),
+            StdString::from(var_length_chars)
+        );
+
+        let null_bytes = "Hello\x00World";
+        let null_bytes_soroban = String::from_str(&env, &null_bytes);
+        assert_eq!(
+            to_std_string(null_bytes_soroban).unwrap(),
+            StdString::from(null_bytes)
+        );
+
+        let escape_char = "Hello\tWorld";
+        let escape_char_soroban = String::from_str(&env, &escape_char);
+        assert_eq!(
+            to_std_string(escape_char_soroban).unwrap(),
+            StdString::from(escape_char)
+        );
+    }
+
+    #[test]
+    fn to_std_string_fails_invalid_utf8_bytes() {
+        let env = Env::default();
+
+        let invalid_sequences = vec![
+            String::from_bytes(&env, &vec![0xF5, 0x90, 0x80]), // not valid utf-8
+            String::from_bytes(&env, &vec![0x00, 0x01, 0x02, 0xC0]), // valid ASCII characters followed by an invalid UTF-8 starting byte
+            String::from_bytes(&env, &vec![0xC0, 0x80, 0xF5, 0x90]), // invalid UTF-8 starting byte followed by valid UTF-8 sequences
+            String::from_bytes(&env, &vec![0xF0, 0x90, 0x80, 0xDF, 0xFB, 0xBF]), // surrogate pairs with invalid charaters "\uD800\uDDFF"
+            String::from_bytes(&env, &vec![0xF4, 0x90, 0x80, 0x80]), // outside the Basic Multilingual Plane
+        ];
+
+        for sequence in invalid_sequences {
+            let result = to_std_string(sequence);
+            assert!(matches!(result, Err(MessageError::InvalidUtf8)));
+        }
+    }
+
+    #[test]
+    #[should_panic]
+    fn soroban_string_copy_to_slice_panics_on_missized_array() {
```

Comment:
this test isn't really relevant since it's testing the behaviour of `copy_into_slice` and not a function that we wrote

Reply:
I realize this. Chris asked me to run a test to confirm that this works as expected, but I can remove.